### PR TITLE
feat(relatedperson): add no related person message and loading indicator

### LINKED ISSUE #1789: Related Persons screen always shows loading when there are no related persons
🐛 Bug Report

When a patient does not have a related person, the related person tab always shows loading. 

## To Reproduce

1. Navigate to a patient without a related person
2. Navigate to the related persons tab
3. See that the related person tab always shows loading

## Expected behavior
A warning message should appear stating that there are no related persons and describe how to add one. 

However, it should show a loading message when it is loading.

### PR DESCRIPTION
**Pull request by Pranav2612000**
Fixes #1789.

**Changes proposed in this pull request:**

- Displays "Loading..." when the component is being loaded.
-Replaces "Loading" with list of related persons(directly) if related persons have been added
- Replaces "Loading..." with "No related persons have been added yet" for patients with no related persons.

### CODE REVIEW

File: src/patients/related-persons/RelatedPersonTab.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ const RelatedPersonTab = (props: Props) => {
   const { permissions } = useSelector((state: RootState) => state.user)
   const [showNewRelatedPersonModal, setShowRelatedPersonModal] = useState<boolean>(false)
   const [relatedPersons, setRelatedPersons] = useState<Patient[] | undefined>(undefined)
+  const [dataFetched, setDataFetched] = useState<boolean>(false)
```
Comment (jackcmeyer) [inline src/patients/related-persons/RelatedPersonTab.tsx]: I think we should use `isLoading` here to stay consistent with other parts of the application. Thoughts?

File: src/patients/related-persons/RelatedPersonTab.tsx [hunk 2]
```diff
@@ -87,14 +90,18 @@ const RelatedPersonTab = (props: Props) => {
...
-              <h1>Loading...</h1>
+            {dataFetched ? (
+              relatedPersons ? (
+                <List>
+                  {relatedPersons.map((r) => (
+                    <ListItem key={r.id}>{r.fullName}</ListItem>
+                  ))}
+                </List>
+                  ) : (
+                <h1>No related persons have been added yet.</h1>
```
Comment (jackcmeyer) [inline src/patients/related-persons/RelatedPersonTab.tsx]: What do you think about using the `Alert` component with the `warning` color?

File: src/patients/related-persons/RelatedPersonTab.tsx [hunk 3]
```diff
@@ -87,14 +90,18 @@ const RelatedPersonTab = (props: Props) => {
...
           <Panel title={t('patient.relatedPersons.label')} color="primary" collapsible>
-            {relatedPersons ? (
-              <List>
-                {relatedPersons.map((r) => (
-                  <ListItem key={r.id}>{r.fullName}</ListItem>
-                ))}
-              </List>
-            ) : (
-              <h1>Loading...</h1>
+            {dataFetched ? (
```
Comment (jackcmeyer) [inline src/patients/related-persons/RelatedPersonTab.tsx]: We'll need tests covering this functionality: 

* test for loading bar while loading
* test for the message when there are no related persons

File: src/patients/related-persons/RelatedPersonTab.tsx (hunk 1 above)
Comment (Pranav2612000) [reply src/patients/related-persons/RelatedPersonTab.tsx]: Will Do.

File: src/patients/related-persons/RelatedPersonTab.tsx (hunk 2 above)
Comment (Pranav2612000) [reply src/patients/related-persons/RelatedPersonTab.tsx]: Yes. It's much better.

File: src/patients/related-persons/RelatedPersonTab.tsx [hunk 4]
```diff
@@ -87,14 +90,18 @@ const RelatedPersonTab = (props: Props) => {
...
+                <List>
+                  {relatedPersons.map((r) => (
+                    <ListItem key={r.id}>{r.fullName}</ListItem>
+                  ))}
+                </List>
+                  ) : (
+                <h1>No related persons have been added yet.</h1>
+                  )
+              ) : (
+                <h1>Loading...</h1>
```
Comment (jackcmeyer) [inline src/patients/related-persons/RelatedPersonTab.tsx]: We can use the `Spinner` component here!

Comment (Pranav2612000) [reply src/patients/related-persons/RelatedPersonTab.tsx]: On it.

**Comment by Pranav2612000** [comment]
Alert


How does this look?

**Comment by tehKapa** [comment]
@HospitalRun/core-maintainers It’s similar to this? https://github.com/HospitalRun/hospitalrun-frontend/pull/1810

**Comment by fox1t** [comment]
There are some conflicts that need to be addressed.

**Comment by Pranav2612000** [comment]
Should I continue working on this, considering there is a similar request for the same issue?
Could you help me in identifying the conflicts that are to be resolved? Thanks

**Comment by tehKapa** [comment]
@Pranav2612000 no. Anyway thanks for the contribution!

**Comment by fox1t** [comment]
This is really unfortunate. In the future, we need to prevent cases like this.

**Pull request closed without merging**